test/unit/data-generator.js

```diff
@@ -0,0 +1,361 @@
+ *
+ * @param {Number} n - Number of tests
+ * @param {Array.<Number>} randomNumbers - A number for each global test,
+ *    even => pass, odd => fail
+ *
+ * @returns {Array.<Test>}
+ */
+function generateNGlobalTests(n, randomNumbers) {
+  var tests = [];
+
+  for (var i = 0; i < n; i++) {
+    if (randomNumbers[i] % 2 === 0) {
+      tests.push(passTest);
+    } else {
+      tests.push(failTest);
+    }
+  }
+
+  return tests;
+}
+
+/*function generateNRandom(n, randomNumbers, passOrFail) {
+  var random = [rootSuite];
+
+  for (var i = 0; i < n; i++) {
+    if (randomNumers[i] % 3 === 0) {
+      random.push(globalSuite);
+    } else if (randomNumbers[i] % 3 === 1) {
+      random.push(*/
+
+describe('Data generator', function() {
+  var runner;
+
+  beforeEach(function() {
+    runner = new EventEmitter();
+  });
+
+  it('should generate only the rootSuite', function(done) {
+    generate(runner, function(data) {
+      expect(data).to.be.deep.equal([rootSuite]);
+
+      done();
+    });
+
+    runner.emit('suite', root);
+    runner.emit('suite end', root);
+    runner.emit('end');
+  });
+
+  it('should generate the rootSuite with random global tests', function(done) {
+    var n = getRandomNumber(),
+        randomNumbers = generateArrayOfRandomNumbers(n);
+
+    generate(runner, function(data) {
+      var tests = generateNGlobalTests(n, randomNumbers),
+          expected = [objectAssign({}, rootSuite, {tests: tests})];
+
+    expect(data).to.be.deep.equal(expected);
+
+      done();
+    });
+
+    runner.emit('suite', root);
+
+    for (var i = 0; i < n; i++) {
+      if (randomNumbers[i] % 2 === 0) {
+        runner.emit('pass', test);
+      } else {
+        runner.emit('fail', test);
+      }
+    }
+
+    runner.emit('suite end', root);
+    runner.emit('end');
+  });
+
+
+  it('should generate n void global suites', function(done) {
+    var randomNumber = getRandomNumber();
+
+    generate(runner, function(data) {
+      var expected = generateNGlobalSuites(randomNumber);
+
+      expect(data).to.be.deep.equal(expected);
+
+      done();
+    });
+
+    runner.emit('suite', root);
+
+    for (var i = 0; i < randomNumber; i++) {
+      runner.emit('suite', suite);
+      runner.emit('suite end', suite);
+    };
+
+    runner.emit('suite end', root);
+    runner.emit('end');
+  });
+
+  it('should generate n void nested suites', function(done) {
+    var randomNumber = getRandomNumber();
+
+    generate(runner, function(data) {
+      var expected = generateNNestedSuites(randomNumber);
+
+      expect(data).to.be.deep.equal(expected);
+
+      done();
+    });
+
+    runner.emit('suite', root);
+
+    for (var i = 0; i < randomNumber; i++) {
+      runner.emit('suite', suite);
+    }
+
+    for (var i = 0; i < randomNumber; i++) {
+      runner.emit('suite end', suite);
+    }
+
+    runner.emit('suite end', root);
+    runner.emit('end');
+  });
+
+  it('should generate n global suites with random tests', function(done) {
```

Comment:
Test name is not very suggestive. You don't want the output to be random — the input is random and the output should respect a certain structure.